gobblin-data-management/src/main/java/org/apache/gobblin/data/management/conversion/hive/source/HiveSource.java

```diff
@@ -495,11 +495,12 @@ private boolean isAvro(Table table) {
   public static void setLineageInfo(ConvertibleHiveDataset convertibleHiveDataset, WorkUnit workUnit,
       SharedResourcesBroker<GobblinScopeTypes> sharedJobBroker)
       throws IOException {
-    String sourceTable =
-        convertibleHiveDataset.getTable().getDbName() + "." + convertibleHiveDataset.getTable().getTableName();
-    DatasetDescriptor source = new DatasetDescriptor(DatasetConstants.PLATFORM_HIVE, sourceTable);
-    source.addMetadata(DatasetConstants.FS_URI,
-        convertibleHiveDataset.getTable().getDataLocation().getFileSystem(new Configuration()).getUri().toString());
+    // Create source dataset descriptor
```

Comment:
I understand the goal here is to provide hdfs based lineage for avro-to-orc conversion instead of hive based lineage. However, to keep the right semantic, `HiveSource` should indicate lineage from `Hive`.  We can do the following to achieve the goal:
- Add `sourceLocation` metadata to the existing hive lineage. `FS_URI` metadata isn't really required but more acts as an example of setting a metadata. Feel free to remove it
- Develop a avro-to-orc specific `DatasetResolver` to convert the hive based lineage to hdfs based.